src/aind_codeocean_utils/codeocean_job.py

```diff
@@ -0,0 +1,495 @@
+"""Module with generic Code Ocean job"""
+import logging
+import time
+from copy import deepcopy
+from datetime import datetime
+from typing import Optional, Union
+
+import requests
+from aind_codeocean_api.codeocean import CodeOceanClient
+from aind_codeocean_api.models.computations_requests import RunCapsuleRequest
+from aind_codeocean_api.models.data_assets_requests import (
+    CreateDataAssetRequest,
+    Source,
+    Sources,
+)
+from aind_data_schema.core.data_description import (
+    DataLevel,
+    datetime_to_name_string,
+)
+
+from aind_codeocean_utils.models.config import (
+    CaptureResultConfig,
+    CodeOceanJobConfig,
+    RegisterDataConfig,
+    RunCapsuleConfig,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class CodeOceanJob:
+    """
+    This class contains convenient methods to register data assets,
+    run capsules, and capture results.
+    """
+
+    def __init__(
+        self,
+        co_client: CodeOceanClient,
+        job_config: Union[dict, CodeOceanJobConfig],
```

Comment:
I think it might be better to rely on the user to do the conversion from dict to `CodeOceanJobConfig`. Pydantic has methods to parse different sources. If the `dict` is from a de-serialized json object, then this requires 
`CodeOceanJobConfig.model_validate(job_config)`
vs. if the user is inputting the dictionary as a set of kwargs.

Reply:
Sounds good! We just need to provide some documentation (maybe directly in the README)